### PR TITLE
Accepting invites using google login

### PR DESCRIPTION
**Pull request by chiragchhatrala**
## Summary by CodeRabbit

- New Features
  - Invite-based sign-up via OAuth, including passing and enforcing invite tokens across redirect, widget, and callback flows.
  - Stateful OAuth redirects with improved security and email-restriction support for providers (e.g., Google, Stripe).
- Refactor
  - Centralized OAuth flows into a single orchestrator; controllers and drivers streamlined.
  - Workspace/role assignment moved to a dedicated service.
- Documentation
  - Updated integration guide to reflect orchestrator, state tokens, intents, and email restrictions.
- Tests
  - Added comprehensive unit and feature tests for OAuth flows, edge cases, invites, and context handling.

### CODE REVIEW

File: client/composables/query/useOAuth.js [hunk 1]
```diff
@@ -101,13 +101,18 @@ export function useOAuth() {
...
+  const guestConnect = (service, redirect = false, additionalData = {}) => {
     const intention = new URL(window.location.href).pathname
 
     return oauthApi.redirect(service, {
       ...redirect ? { intention } : {},
+      ...additionalData
     })
       .then((data) => {
+        // If we have invite_token in additionalData store it in localStorage
+        if (additionalData.invite_token) {
```
Comment (JhumanJ) [inline client/composables/query/useOAuth.js:113]: Why do we need to store this?

File: client/pages/oauth/[provider]/callback.vue [hunk 2]
```diff
@@ -42,11 +42,19 @@ const handleCallback = async () => {
   const provider = route.params.provider
   
   try {
+    let payloadData = {
+      code: route.query.code,
+      utm_data: $utm.value
+    }
+    // Get invite token from localStorage if it was stored during OAuth initiation
+    const inviteToken = localStorage.getItem('oauth_invite_token')
```
Comment (JhumanJ) [inline client/pages/oauth/[provider]/callback.vue:56]: Why do we need to send it again here? We should cache it when creating the initial login attempt as we already did? So no need to do it again on client side?

File: api/app/Http/Controllers/Auth/OAuthController.php [hunk 3]
```diff
@@ -219,14 +266,42 @@ private function getIntentFromContext(): string
...
             $driver->setScopes($scopes);
         }
 
         return $driver->getRedirectUrl();
     }
+
+    /**
+     * Validate invite token and return invited email
+     */
+    private function validateInviteToken(string $inviteToken): string
```
Comment (JhumanJ) [inline api/app/Http/Controllers/Auth/OAuthController.php]: this should probably also be in workspace invite service I think. Same as for `validateInviteEmailRestrictions`

File: api/app/Http/Controllers/Auth/OAuthController.php [hunk 4]
```diff
@@ -53,20 +63,27 @@ public function redirect(Request $request, string $provider)
...
-                'utm_data' => $request->input('utm_data')
-            ], now()->addMinutes(5));
+                'utm_data' => $request->input('utm_data'),
+                'invited_email' => $invitedEmail
+            ]);
+        }
+
+        // Set email restrictions on the driver if it's Google and we have an invited email
+        $driver = $providerService->getDriver();
+        if ($provider === 'google' && $invitedEmail && $driver instanceof OAuthGoogleDriver) {
```
Comment (JhumanJ) [inline api/app/Http/Controllers/Auth/OAuthController.php]: This is the kind of thing we should avoid, we want to keep this generic and easy to use

File: api/app/Http/Controllers/Auth/OAuthController.php [hunk 5]
```diff
@@ -76,7 +93,14 @@ public function callback(string $provider)
         // Determine intent from cached context
         $intent = $this->getIntentFromContext();
 
-        return $this->handleIntent($intent, $providerService, $userData);
+        // For auth intent with invite token, validate and set cache context
+        $invitedEmail = null;
+        $inviteToken = $request->input('invite_token');
+        if ($inviteToken && $intent === self::INTENT_AUTH) {
+            $invitedEmail = $this->validateInviteToken($inviteToken);
```
Comment (JhumanJ) [inline api/app/Http/Controllers/Auth/OAuthController.php]: Controller should not have specific code for invites..

**Comment by chiragchhatrala** [comment]
All changes done
